pkg/segment/structs/segstructs.go

```diff
@@ -721,6 +723,34 @@ func GetMeasureAggregatorStrEncColumns(measureAggs []*MeasureAggregator) []strin
 	return columns
 }
 
+func (ss *StringStats) MergeMinStrStats(other *StringStats) {
+	if ss.Min.Dtype == utils.SS_INVALID {
+		ss.Min = other.Min
+	}
```

Comment:
nit: It might make sense to add
```
if ss == nil {
    return other
}
```
at the top of this function. I think that could also simplify some of the logic in GetSegMin (and GetSegMax)

But you can ignore this if you like your way better

Reply:
Wouldn't it give panic if you call function on a nil value ?

Reply:
No (see https://go.dev/play/p/kUK7OcqIjQN)

Golang is a little odd in that regard; this is basically how Go handles what other functions call static methods

Reply:
Well this is something new :), will update it.

Reply:
I just checked, we are not returning anything over here and this condition already handled in the merging of segstats [here](https://github.com/siglens/siglens/blob/b70392af3b4e6c892cb43e51447a8c10dd8f8cb7/pkg/segment/structs/segstructs.go#L740).

Reply:
I have kept is as it is, if needed we can cleanup in the future.